transport/http/client.go

```diff
@@ -121,3 +121,16 @@ func WrapLogClient(logger interface{ Logf(string, ...interface{}) }, client Clie
 		return resp, nil
 	})
 }
+
+// NopClient provides a client that ignores the request, and returns a empty
+// successful HTTP response value.
+type NopClient struct{}
```

Comment:
1. Is this required for testing? or why should this be in `transport/http/client.go`?

2. If for testing, should this be modified to:

```
type NopClient struct{
	StatusCode int
	Header []http.Header
	Body   io.ReadCloser
}

```

That way we can create clients which can return error case and stuff. 

The Do implementation that you have below, can be used for the default returned values in case struct member isn't populated.

Reply:
This is required for the presign URL behavior, not tests.  The purpose of it is to have a no-op HTTP client that will do nothing but return success. Basically to prevent a middleware stack making a request with a presigned URL flow.